File: dziennik-online/app/controllers/teacher/school_classes_controller.rb
class Teacher::SchoolClassesController < ApplicationController

  def index
    @school_classes = current_teacher.school_classes
    @school_class = current_teacher.school_classes.new
    @semester = Semester.new
  end

  def create
    @school_class = current_teacher.school_classes.build(params[:school_class])
    @school_class.save ? redirect_to(:action => "index") : render(:action => "index")
  end

  def destroy
    @school_class = current_teacher.school_classes.find(params[:id])
    @school_class.destroy
    redirect_to(:action => "index")
  end

end

File: dziennik-cakephp/app/Controller/SchoolClassesController.php
<?php
class SchoolClassesController extends AppController {

  public $name = 'SchoolClasses';
  public $helpers = array('SchoolClass');

  function beforeFilter() {
    parent::beforeFilter();
    if ($this->action == 'teacher_view' || $this->action == 'teacher_delete') {
      $this->classAuth();
    }
  }

  private function classAuth() {
    if (!$this->SchoolClass->findByIdAndTeacherId($this->params['pass']['0'], $this->currentUser('id'))) {
      $this->Session->setFlash('Brak dostępu.', 'flash_error');
      $this->redirect($this->referer());
    }
  }

  function teacher_create() {
    if ($this->request->is('post')) {
      $this->SchoolClass->create();
      if ($this->SchoolClass->save($this->request->data)) {
        $this->Session->setFlash('Dodano nową klasę.', 'flash_success');
        $this->redirect(array('controller' => 'school_classes', 'action' => 'index'));
      }
    }
  }

  function teacher_index() {
    $this->set('classes', $this->SchoolClass->findAllByTeacherId($this->currentUser('id')), array(), array('Semesters.id' => 'ASC'));
    $this->set('semester_actual', $this->currentUser('semester_id'));
    $this->set('class_actual', $this->currentUser('class_id'));
  }

  function teacher_delete() {
    if ($this->SchoolClass->delete($this->params['pass']['0'])) {
      $this->Session->setFlash('Usunięto klasę pomyślnie.', 'flash_success');
      $this->redirect('/teacher/school_classes');
    }
  }

}
?>
